django/contrib/admin/options.py

```diff
@@ -281,6 +270,23 @@ def formfield_for_manytomany(self, db_field, request, **kwargs):
             form_field.help_text = format_lazy('{} {}', help_text, msg) if help_text else msg
         return form_field
 
+    def get_widget_for_related_field(self, db_field, formfield, request):
+        """
+        Return a wrapped widget instance for a related field
+        """
+        related_modeladmin = self.admin_site._registry.get(db_field.remote_field.model)
+        wrapper_kwargs = {}
+        if related_modeladmin:
+            wrapper_kwargs.update(
+                can_add_related=related_modeladmin.has_add_permission(request),
+                can_change_related=related_modeladmin.has_change_permission(request),
+                can_delete_related=related_modeladmin.has_delete_permission(request),
+                can_view_related=related_modeladmin.has_view_permission(request),
+            )
+        return widgets.RelatedFieldWidgetWrapper(
```

Comment:
After implement the full functionality it would be handier to have an helper that just returns the instance and given formfield, db_field and wrapper_args.